src/Analyzers/Core/CodeFixes/MakeMemberStatic/AbstractMakeMemberStaticCodeFixProvider.cs

```diff
@@ -29,22 +50,90 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
             return Task.CompletedTask;
         }
 
-        protected sealed override Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor,
+        private static bool TryGetIndexOfRawKind(SyntaxTokenList modifiers, int rawKind, out int index)
+        {
+            for (var i = 0; i < modifiers.Count; i++)
+            {
+                if (modifiers[i].RawKind == rawKind)
+                {
+                    index = i;
+                    return true;
+                }
+            }
+
+            index = -1;
+            return false;
+        }
+
+        private SyntaxTokenList InsertStaticModifier(SyntaxTokenList modifiers, string preferredOrder)
+        {
+            if (modifiers.Count == 0)
+            {
+                return new SyntaxTokenList(StaticModifier);
+            }
+
+            var order = preferredOrder.Split(',').Select(m => m.Trim()).ToArray();
```

Comment:
so this will parse this every time, causing allocations.  can you follow the pattern of TryGetOrComputePreferredOrder in AbstractOrderModifiersCodeFixProvider so we can cache the last parse and use that instead.